### PR TITLE
Add moonglum

### PR DESCRIPTION
**Pull request by moonglum**
I added myself to the project and it worked :smile: The result was a VMWare Box :smile: 

In order to do that, I removed the `common` playbook and added the role to the `yoshida` playbook. Then I copied my playbook from my dotfiles into the directory. I removed the tasks that need directories from my computer mounted into the machine. Then I ran the build script :wink: 

How can we run this now? Is this on Atlas or on Codeship?

### CODE REVIEW

File: scripts/centos/cleanup.sh [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/bin/sh -eux
 
 if [ -s /etc/oracle-release ]; then
-  distro = 'oracle'
+  distro='oracle'
```
Comment (bascht) [inline scripts/centos/cleanup.sh:4]: So much for that OCD. :D

Comment (bascht) [inline scripts/centos/cleanup.sh:4]: Oh. Wait. This is bash, isn't it?
Could be (somewhat) significant whitespace.

Comment (moonglum) [inline scripts/centos/cleanup.sh:4]: It is bash :wink: And also: I did not write this – this is synced from the Bento repository :wink:

Comment (bascht) [inline scripts/centos/cleanup.sh:4]: I stand corrected. :bow:

**Comment by moonglum** [comment]
Success! I switched from 15.10 to 15.04, now I had my first successful `vagrant up && vagrant ssh`. One thing that confuses me:

I tried to provision it with:

``` ruby
config.vm.provision "ansible_local" do |ansible|
  ansible.playbook = "playbooks/main.yml"
end
```

It tells me:

```
The Ansible software could not be found! Please verify
that Ansible is correctly installed on your guest system.

If you haven't installed Ansible yet, please install Ansible
on your Vagrant basebox, or enable the automated setup with the
`install` option of this provisioner. Please check
https://docs.vagrantup.com/v2/provisioning/ansible_local.html
for more information.
```

I thought you used `ansible_local` in Packer? Shouldn't it still be installed? Oddly, `ansible.install = true` doesn't change that :confused: Apart from that it seems to be working :smile:

File: Rakefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "json"
 require "yaml"
 
-DEFAULT_BASE_OS = "ubuntu-15.10"
+DEFAULT_BASE_OS = "ubuntu-15.04"
```
Comment (code-later) [inline Rakefile:4]: Why did you used this version? Had you problems with 15.10?

**Comment by code-later** [comment]
Great this worked out for you @moonglum. But I have no idea why Vagrant could not find the installed `ansible`. Did you checked it was really installed on the box?

Regarding your other question: This is build on Atlas. You need to create a new build and configure the `DEVELOPER_PLAYBOOK` variable accordingly. You can look and the build for my VM for this. It should build the box as soon as you merge into master.

For testing purpose you can always push to Atlas manually and trigger a build. But you need to have a build configured beforehand. If you want to do this, please add a Rake task like the following:

``` ruby
desc "Push manually to Atlas"
task :push => :assemble do
  puts "Push development configuration to build at Atlas"
  system "packer push -name yakshed/#{ENV.fetch('DEVELOPER')} base.json"
end
```

I'm pretty sure you have to check for the required variables to pass to the `push` command, not sure what will really be needed ;-)

File: Rakefile (hunk 2 above)
Comment (moonglum) [inline Rakefile:4]: Yes :sob: I had a long debugging session via screen sharing with @bascht yesterday. The result was that there was something really fucked up in the network preferences of the box. I had the gut feeling that there might be something wrong in the 15.10 version, and that gut feeling was right: I created the box with 15.10: Could not connect to it, network was fucked up. I ONLY changed this line, generated a new base.json and everything worked.

**Comment by moonglum** [comment]
I tried another `vagrant up` run just now. `ansible` is installed in the box. So I tried running it via `ansible` instead of `ansible_local`. The result is:

```
fatal: [default] => Authentication or permission failure.  In some cases, you may have been able to authenticate and did not have permissions on the remote directory. Consider changing the remote temp path in ansible.cfg to a path rooted in "/tmp". Failed command was: mkdir -p $HOME/.ansible/tmp/ansible-tmp-1453465318.49-279836142813079 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1453465318.49-279836142813079 && echo $HOME/.ansible/tmp/ansible-tmp-1453465318.49-279836142813079, exited with result 1: mkdir: cannot create directory ‘/home/vagrant/.ansible/tmp/ansible-tmp-1453465318.49-279836142813079’: Permission denied
```

So I looked at the permissions. `/home/vagrant/.ansible` is owned by root, and only root can write to it. I don't entirely get why this is a problem as the playbook has a `Sudo: True`. But it's definitely a difference to my current box where the folder is owned by `vagrant`. I changed the owner to vagrant, now the `ansible` provider works (and is so fucking fast, YEAH!!!). But `ansible_local` still fails :confused: 

Cool, so I will just copy your configuration on Atlas and adjust the names and then it will build my box as soon as we merge this. Correct? :dancer:

File: Rakefile (hunk 2 above)
Comment (code-later) [inline Rakefile:4]: :-( okay then…

Comment (moonglum) [inline Rakefile:4]: We'll try updating to 15.10 later :smile:

**Comment by code-later** [comment]
Why do you want to run Ansible during `vagrant up` anyway?

And yes, as soon as we merge Atlas will build this, but I suggest you test this first by manually pushing ;-)

As a bonus round you _could_ use Codeship to push to Atlas. This would get rid of the `assemble` task and we could use bento as a git submodule.

**Comment by moonglum** [comment]
My ansible playbook:

``` yaml
---
- name: Install the required packages
  hosts: all
  sudo: True
  vars:
    marks:
      c: /home/vagrant/Code
  tasks:
    - shell: ls -1 /home/vagrant/Code/dotfiles/tilde
      register: dotfiles
    - name: Link Dotfiles
      file: path=/home/vagrant/.{{ item }} src=/home/vagrant/Code/dotfiles/tilde/{{ item }} owner=vagrant group=vagrant state=link
      with_items: dotfiles.stdout_lines
    - name: Create the marks directory
      file: path=/home/vagrant/.marks state=directory
    - name: Configure the marks
      file: path=/home/vagrant/.marks/{{ item.key }} state=link src={{ item.value }}
      with_dict: marks
```

Those tasks can only be executed when my directories are mounted into the box :smile: 

Manual Push: Ok, will do :wink: 
Codeship: One task at a time, my friend!

**Comment by moonglum** [comment]
This is ready for merge, @railsbros-dirk :smile: You can see the build in our Atlas org :wink: I did my first successful `vagrant up` with that box, and it was super fast :rocket: I think we have the follow up tasks:
- Hook everything up to Codeship and build the base.json there. Push from Codeship to Atlas.
- Extract roles from @railsbros-dirk and @moonglum's playbook, so there's less duplication. Also we might steal stuff from each other along the way :smile: 
- Update to 15.10 later, make the codename of the release available in the playbooks (for things like adding ppas)

Should I create issues for those?

**Comment by moonglum** [comment]
Merge, @railsbros-dirk, merge!

**Comment by code-later** [comment]
Great work @moonglum

**Comment by moonglum** [comment]
Thanks, @railsbros-dirk

**Pull request merged**